### PR TITLE
OGM-1411 Backporting MongoDB native query with pagination on aggregate

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/OGM-1411

### CODE REVIEW

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java [hunk 1]
```diff
@@ -151,6 +145,12 @@
 import com.mongodb.client.model.UpdateOptions;
 import com.mongodb.client.result.DeleteResult;
 import com.mongodb.client.result.UpdateResult;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.parboiled.Parboiled;
+import org.parboiled.errors.ErrorUtils;
+import org.parboiled.parserunners.RecoveringParseRunner;
+import org.parboiled.support.ParsingResult;
```
Comment (DavideD) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java]: No need to change in this PR but make sure in the future to not reorder the imports unless for good reasons

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/981

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java (hunk 1 above)
Comment (DavideD) [reply mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java]: I've reverted it before merging

Comment (fax4ever) [reply mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java]: ok thank you

**Pull request closed without merging**